homeassistant/components/mqtt/binary_sensor.py

```diff
@@ -114,6 +116,34 @@ def __init__(self, hass, config, config_entry, discovery_data):
 
         MqttEntity.__init__(self, hass, config, config_entry, discovery_data)
 
+    async def async_added_to_hass(self) -> None:
+        """Recover active entities with exire trigger."""
+        await super().async_added_to_hass()
+        if (
+            (expire_after := self._config.get(CONF_EXPIRE_AFTER)) is not None
+            and expire_after > 0
+            and (last_state := await self.async_get_last_state()) is not None
+            and last_state != STATE_UNAVAILABLE
+        ):
+            self._expired = False
+            self._state = last_state.state
+            expiration_at = dt_util.utcnow() + timedelta(seconds=expire_after)
+            self._expiration_trigger = async_track_point_in_utc_time(
+                self.hass, self._value_is_expired, expiration_at
+            )
+            _LOGGER.debug("State recovered after reload for %s", self.entity_id)
```

Comment:
Maybe print the remaining time of the expiration trigger too

Reply:
Done